rem/remApp.swift

```diff
@@ -380,13 +402,14 @@ func drawStatusBarIcon(rect: CGRect) -> Bool {
                 }
                 
                 // Might as well only check if the applications are the same, otherwise obviously different
-                if activeApplicationName != lastActiveApplication || displayImageChangedFromLast(imageData: imageData) {
+                if activeApplicationName != lastActiveApplication || lastDisplayID != displayID || displayImageChangedFromLast(imageData: imageData) {
                     lastImageData = imageData;
                     lastActiveApplication = activeApplicationName;
+                    lastDisplayID = displayID;
                     
                     let frameId = DatabaseManager.shared.insertFrame(activeApplicationName: activeApplicationName)
                     
-                    if settingsManager.settings.onlyOCRFrontmostWindow {
+                    if settingsManager.settings.onlyOCRFrontmostWindow && displayID == CGMainDisplayID() {
```

Comment:
is this still applicable?

Reply:
shouldn't it be `lastDisplayID`?

Reply:
Yeah so for mouse mode it is to make sure the image isn't cropped when the active window is on the main display but the recorded display is not the main display.

For active window mode it doesn't matter (for my set up at least) since the cropping step will fail because the window will have negative indexes (Also the frame would be wrong either way). I briefly tried to recover this functionality for the external monitors but ran into the funky window.frame behaviors. Did you think that the cropping was working for external monitors previously?

Reply:
> funky window frame behaviors

Is the `backingScalingFactor` being properly applied to the window frames? that's usually the culprit.

I assume the window frames need to be repositioned based on the monitor.

I think there's a larger issue here.

So - should checking the active mouse mode uncheck the only ocr frontmost window setting?

Reply:
(Seems like they can't be used together)

Reply:
Yeah I mean feasibly it could work when both the mouse and active window are on display but would require some more complex handling (that we can't even currently do without consistently identifying the display of the active window).

Also, I think the reality is for people with multiple monitors most of the time the front window will be the entire display so I don't think the feature is a big save. I'm going to look at the window.frame problem with some fresh eyes and try the `backingScalingFactor` again but assuming that doesn't work one option could be ignoring `settingsManager.settings.onlyOCRFrontmostWindow` when there are multiple displays. Otherwise the whole thing gets a bit funky since the window is only ever cropped on the main display.

Reply:
With every `window.frame` exploration the mystery deepens. So It seems like it is possible to match up a `window.frame` to a screen location by making the y coordinate of the screen negative. However, I would be very hesitant to assume this is the case for all set ups.

Here is an example where I had all 3 windows on fullscreen:

App: Xcode
window.frame: (712.0, -1056.0, 1920.0, 54.0)
Screen: VG245, Display ID: 5, Coordinates: x=712.0, y=982.0, Width=1920.0, Height=1080.0
backingScaleFactor: 1.0

App: Firefox
window.frame: (-1208.0, -1080.0, 1920.0, 72.0)
Screen: HP 24mh, Display ID: 3, Coordinates: x=-1208.0, y=982.0, Width=1920.0, Height=1080.0
backingScaleFactor: 1.0

App: Messages
window.frame: (0.0, 37.0, 1512.0, 68.0)
Screen: Built-in Retina Display, Display ID: 1, Coordinates: x=0.0, y=0.0, Width=1512.0, Height=982.0
backingScaleFactor: 2.0

"Funky" behavior: 
1) Weird very small heights (sometimes) when in full screen. Ran two back to back tests without changing the code or moving the window and one had a height of 72 and one had 1080. Strange the widths always seem to be reasonable 🤷
2) The `window.frame` doesn't change when the window is manipulated. I tried both moving it around and resizing it.
3) I have a terminal open on each monitor and it seems to always return the `window.frame` of the terminal on the main monitor even if I am actively using the terminal on an external monitor.

From my perspective, the cleanest solution is to ignore the `settingsManager.settings.onlyOCRFrontmostWindow` setting when there are multiple displays. Based on some online research and my belief that rewind just uses mouse location and no window based OCR optimization, I'm not entirely sure a robust solution exists given the current APIs. Also for most users with multiple monitors, I doubt it will be much of a compute save anyways.

What do you think?

Reply:
only running ocr on the current window can be a pretty large savings **especially** for people using multiple monitors. if i have a 4k monitor, i can easily have 4-5 open windows with text. only doing 1 is a big deal. running rem 24/7 could/can be expensive from an energy perspective, so always working to reduce it as much as possible.